src/main/java/uk/gov/hmcts/reform/sscs/util/PdfRequestUtils.java

```diff
@@ -0,0 +1,176 @@
+    }
+
+    public static PreSubmitCallbackResponse<SscsCaseData> processPostHearingRequestPdfAndSetPreviewDocument(
+        String userAuthorisation,
+        SscsCaseData sscsCaseData,
+        PreSubmitCallbackResponse<SscsCaseData> response,
+        GenerateFile generateFile,
+        String templateId
+    ) {
+        return processRequestPdfAndSetPreviewDocument(PdfType.POST_HEARING, userAuthorisation, sscsCaseData, response, generateFile, templateId);
+    }
+
+    public static PreSubmitCallbackResponse<SscsCaseData> processRequestPdfAndSetPreviewDocument(
+        PdfType pdfType,
+        String userAuthorisation,
+        SscsCaseData sscsCaseData,
+        PreSubmitCallbackResponse<SscsCaseData> response,
+        GenerateFile generateFile,
+        String templateId
+    ) {
+        log.debug("Executing processRequestPdfAndSetPreviewDocument for caseId: {}", sscsCaseData.getCcdCaseId());
+
+        ADDITIONAL_REQUEST_DETAILS.append("Date request received: ").append(LocalDate.now().format(DATE_TIME_FORMATTER)).append("\n");
+        StringBuilder pdfUrlBuilder = new StringBuilder();
+        switch (pdfType) {
+            case POST_HEARING:
+                pdfUrlBuilder.append(getPostHearingDocumentType(sscsCaseData).getLabel());
+                pdfUrlBuilder.append(" from FTA"); // TODO SSCS-10759 make dynamic for the uploading party
+                handlePostHearing(sscsCaseData);
+                break;
+            case POSTPONEMENT:
+                pdfUrlBuilder.append("Postponement Request");
+                handlePostponement(sscsCaseData);
+                break;
+            default:
+                throw new IllegalArgumentException("Unsupported event type for processRequestPdfAndSetPreviewDocument: " + pdfType);
+        }
+        pdfUrlBuilder.append(".pdf");
+
+        if (isBlank(requestDetails)) {
+            response.addError(String.format("Please enter request details to generate a %s document", pdfType.toString().toLowerCase()));
+            return response;
+        }
+
+        DocumentLink previewDocument = getPreviewDocument(pdfUrlBuilder.toString(), userAuthorisation, generateFile, templateId, title);
+
+        switch (pdfType) {
+            case POST_HEARING:
+                sscsCaseData.getDocumentStaging().setPreviewDocument(previewDocument);
+                break;
+            case POSTPONEMENT:
+                sscsCaseData.getPostponementRequest().setPostponementPreviewDocument(previewDocument);
+                break;
+            default:
+                throw new IllegalArgumentException("Unsupported event type for processRequestPdfAndSetPreviewDocument: " + pdfType);
+        }
+
+        return response;
+    }
+
+    private static void handlePostponement(SscsCaseData sscsCaseData) {
+        requestDetails = sscsCaseData.getPostponementRequest().getPostponementRequestDetails();
```

Comment:
`sscsCaseData.getPostponementRequest()` used 3x
Put it to seperate var

Reply:
done